h5py/h5d.pyx

```diff
@@ -395,3 +395,43 @@ cdef class DatasetID(ObjectID):
             """
             H5Drefresh(self.id)
 
+
+    IF HDF5_VERSION >= (1, 8, 11):
+
+        def write_direct_chunk(self, offsets, data, filter_mask=0, PropID dxpl=None):
+            """
+            Direct chunk write to dataset
+            """
+
+            cdef hid_t dset_id
+            cdef hid_t dxpl_id
+            cdef hid_t space_id = 0
+            cdef uint32_t filters
+            cdef hsize_t *offset
```

Comment:
All pointers should be explicitly initialized to NULL, especially if you'll be freeing them later.